conduit/pipeline.go

```diff
@@ -0,0 +1,286 @@
+package conduit
+
+import (
+	"context"
+	"fmt"
+	"github.com/algorand/go-algorand-sdk/encoding/json"
+	"github.com/algorand/indexer/plugins"
+
+	log "github.com/sirupsen/logrus"
+	"gopkg.in/yaml.v3"
+	"io/ioutil"
+	"path/filepath"
+
+	"github.com/algorand/go-algorand/util"
+	"github.com/algorand/indexer/data"
+	"github.com/algorand/indexer/exporters"
+	"github.com/algorand/indexer/importers"
+	"github.com/algorand/indexer/processors"
+)
+
+const autoLoadParameterConfigName = "conduit.yml"
+
+// PipelineConfig stores configuration specific to the conduit pipeline
+type PipelineConfig struct {
+	ConduitConfig *Config
+
+	PipelineLogLevel string `yaml:"LogLevel"`
+	// Store a local copy to access parent variables
+	Importer   map[string]interface{}   `yaml:"Importer"`
+	Processors []map[string]interface{} `yaml:"Processors"`
+	Exporter   map[string]interface{}   `yaml:"Exporter"`
+}
+
+// Valid validates pipeline config
+func (cfg *PipelineConfig) Valid() error {
+	if cfg.ConduitConfig == nil {
+		return fmt.Errorf("conduit configuration was nil")
+	}
+
+	if _, err := log.ParseLevel(cfg.PipelineLogLevel); err != nil {
+		return fmt.Errorf("pipeline log level (%s) was invalid: %w", cfg.PipelineLogLevel, err)
+	}
+
+	if len(cfg.Importer) == 0 {
+		return fmt.Errorf("importer configuration was empty")
+	}
+
+	if len(cfg.Exporter) == 0 {
+		return fmt.Errorf("exporter configuration was empty")
+	}
+
+	return nil
+}
+
+// MakePipelineConfig creates a pipeline configuration
+func MakePipelineConfig(logger *log.Logger, cfg *Config) (*PipelineConfig, error) {
+	if cfg == nil {
+		return nil, fmt.Errorf("empty conduit config")
```

Comment:
Could you use the pattern we use elsewhere with the error string starting with the function name? That has been very helpful for debugging.